### PR TITLE
Add SVG icon loader

### PR DESCRIPTION
**Pull request by zehfernandes**
I saw in the features list the item _Generating icon font from SVGs_ 

Have many articles about what is the best: "Icon Fonts vs. SVG sprite"
- https://css-tricks.com/icon-fonts-vs-svg/
- https://www.sitepoint.com/icon-fonts-vs-svg-debate/

Personally, I prefer SVG and I think is the correct use of it. Symbols nodes with SVG sprite are more versatile, easy to align and very simple to maintain. 

So, what this pull request does:
- Create an SVG sprite using the loader `svg-sprite-loader` just with the SVG files in the `icons` folder
- Icon component to insert the icon using the SVG file name as reference `<Icon glyph='triangle' />`
- Add examples in the Dashboard page

### CODE REVIEW

**Comment by Stanko** [comment]
@zehfernandes This is awesome, thank you! I totally agree this is better approach than icon fonts.

But I'll be a bit nitpicky, it seems like your IDE changed code formatting on few places. So I would really appreciate if you change it before we merge it.

File: source/js/views/Dashboard/index.jsx [hunk 1]
```diff
@@ -2,7 +2,9 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { testAction, testAsync } from 'actions/app';
+import Icon from '../../components/Global/Icon';
```
Comment (Stanko) [inline source/js/views/Dashboard/index.jsx]: you can use  `import Icon from 'components/Global/Icon';`, as project is set up to use `js` folder as one of the context points.

File: source/js/views/Dashboard/index.jsx [hunk 2]
```diff
@@ -40,34 +42,25 @@ export default class Dashboard extends Component {
...
 
   render() {
-    const {
-      asyncData,
-      asyncError,
-      asyncLoading,
-      counter,
-    } = this.props;
-
+    const { asyncData, asyncError, asyncLoading, counter } = this.props;
```
Comment (Stanko) [inline source/js/views/Dashboard/index.jsx]: This is one of the things I think you IDE did automatically, let's keep it multiline, it is easier to track changes on git that way.

File: source/js/views/Dashboard/index.jsx [hunk 3]
```diff
@@ -40,34 +42,25 @@ export default class Dashboard extends Component {
...
     return (
       <div className='Dashboard'>
         <h2>Examples</h2>
         <hr />
         <div>
           <h3>Synchronous action</h3>
-          <p>{ counter }</p>
+          <p>{counter}</p>
```
Comment (Stanko) [inline source/js/views/Dashboard/index.jsx]: IDE again, add spaces back inside the curly braces. We have setup `react/jsx-curly-spacing` but it doesn't catch blocks, only jsx attributes. 

It is known bug though
https://github.com/yannickcr/eslint-plugin-react/issues/857

File: source/js/views/Dashboard/index.jsx [hunk 4]
```diff
@@ -40,34 +42,25 @@ export default class Dashboard extends Component {
...
-          { asyncLoading && <p>Loading...</p> }
-          { asyncError && <p>Error: { asyncError }</p> }
-          <button
-            disabled={ asyncLoading }
-            onClick={ this.handleAsyncButtonClick }
-          >
+          <p>{asyncData}</p>
+          {asyncLoading && <p>Loading...</p>}
+          {asyncError && <p>Error: {asyncError}</p>}
+          <button disabled={ asyncLoading } onClick={ this.handleAsyncButtonClick }>
```
Comment (Stanko) [inline source/js/views/Dashboard/index.jsx]: Formatting - add spaces back to curly braces, keep attributes multiline.

File: webpack.config.js [hunk 5]
```diff
@@ -39,10 +42,7 @@ const plugins = [
     options: {
       postcss: [
         autoprefixer({
-          browsers: [
-            'last 3 version',
-            'ie >= 10',
-          ],
+          browsers: ['last 3 version', 'ie >= 10'],
```
Comment (Stanko) [inline webpack.config.js]: Let's keep this one, as I don't think it will change soon.

File: webpack.config.js [hunk 6]
```diff
@@ -90,40 +102,33 @@ if (isProduction) {
...
+      use: 'css-loader!postcss-loader!sass-loader',
+    }),
+  });
 } else {
   // Development plugins
-  plugins.push(
-    new webpack.HotModuleReplacementPlugin(),
-    new DashboardPlugin()
-  );
+  plugins.push(new webpack.HotModuleReplacementPlugin(), new DashboardPlugin());
```
Comment (Stanko) [inline webpack.config.js]: Formatting - keep it multiline please

File: webpack.config.js [hunk 7]
```diff
@@ -90,40 +102,33 @@ if (isProduction) {
...
+      // Using source maps breaks urls in the CSS loader
+      // https://github.com/webpack/css-loader/issues/232
+      // This comment solves it, but breaks testing from a local network
+      // https://github.com/webpack/css-loader/issues/232#issuecomment-240449998
+      // 'css-loader?sourceMap',
+      'css-loader',
+      'postcss-loader',
+      'sass-loader?sourceMap',
+    ],
+  });
```
Comment (Stanko) [inline webpack.config.js]: Please revert formatting.

File: webpack.config.js [hunk 8]
```diff
@@ -141,11 +146,14 @@ module.exports = {
...
-      path.resolve(__dirname, 'node_modules'),
-      jsSourcePath,
+    extensions: [
+      '.webpack-loader.js',
+      '.web-loader.js',
+      '.loader.js',
+      '.js',
+      '.jsx',
     ],
+    modules: [path.resolve(__dirname, 'node_modules'), jsSourcePath],
```
Comment (Stanko) [inline webpack.config.js]: Formatting - please revert `modules` to multiline. I love that `extensions` are now multiline as well.

**Comment by zehfernandes** [comment]
Sorry about the formatting changes.
I supposed the prettier plugin was using the eslint's rules to format. But not :(

My last commit fixed that. 👍

**Comment by oliverdore-work** [comment]
@zehfernandes I made that assumption about Prettier and ESLint recently too. Easily done.

**Pull request merged**